script/testing/util/common.py

```diff
@@ -22,55 +26,83 @@ def run_command(command,
                          cwd=cwd)
 
     while p.poll() is None:
-        if stdout == subprocess.PIPE:
-            out = p.stdout.readline()
-            if out:
-                LOG.info(out.decode("utf-8").rstrip("\n"))
+        if printable:
+            if stdout == subprocess.PIPE:
+                out = p.stdout.readline()
+                if out:
+                    LOG.info(out.decode("utf-8").rstrip("\n"))
 
     rc = p.poll()
     return rc, p.stdout, p.stderr
 
 
-def check_port(port):
+def print_output(filename):
+    """ Print out contents of a file """
+    with open(filename) as file:
+        lines = file.readlines()
+        for line in lines:
+            LOG.info(line.strip())
+
+
+def format_time(timestamp):
+    return datetime.datetime.fromtimestamp(timestamp).strftime(
+        "%Y-%m-%d %H:%M:%S")
+
+
+def kill_pids_on_port(port):
+    for proc in psutil.process_iter():
+        try:
+            for conns in proc.connections(kind="inet"):
+                if conns.laddr.port == port:
+                    LOG.info(
+                        "Killing existing server instance listening on port {} [PID={}], created at {}"
+                        .format(port, proc.pid,
+                                format_time(proc.create_time())))
+                    proc.send_signal(signal.SIGKILL)
+        except psutil.ZombieProcess:
+            LOG.info("Killing zombie process [PID={}]".format(proc.pid))
+            proc.parent().send_signal(signal.SIGCHLD)
+
+
+def get_pids_on_port(port):
     """Get the list of PIDs (if any) listening on the target port"""
```

Comment:
Again, adding warning/check whether you are running as root.

Reply:
done